### DISCUSSION TITLE
TypeError: Object of type PosixPath is not JSON serializable

### DISCUSSION THREAD

**Discussion by sevetseh28** [pydantic/pydantic#10235] [Question] [answered]
Hi. I am trying to serialise a pydantic model that has Path fields, but I get the error:
`TypeError: Object of type PosixPath is not JSON serializable`

I find it strange that it doesnt just use the `str` representation of it by default. Surprisingly, I couldn't find any related thread or docs either.
Any ideas?
I've tried creating a `CustomBaseModel` like so:

```python
class CustomBaseModel(BaseModel):
    model_config = ConfigDict(
        json_encoders={
            Path: lambda v: str(v),
            PosixPath: lambda v: str(v),
        }
    )

```
but that didn't work either.

Thanks!

**Comment by Viicos** [comment]
Could you please share how you are serializing your model?

**Comment by ecly** [comment]
@sevetseh28 which version of pydantic are you using? 

I've tested with 2.5.3 and the newest on pypi (2.8.2) for both of which both `PosixPath` and `Path` work for me, without any custom JSON encoders. I.e. a full example that works for me would be:

```python
from pathlib import Path, PosixPath
from pydantic import BaseModel, ConfigDict


class CustomBaseModel(BaseModel):
    model_config = ConfigDict()
    path: Path
    posix_path: PosixPath


model = CustomBaseModel(path=Path("/Users/ecly"), posix_path=PosixPath("/Users/ecly"))
print(model.model_dump_json())
from_json = model.model_validate({"path": "/Users/ecly", "posix_path": "/Users/ecly"})
print(from_json)
```

Alternatively, perhaps you have a runnable example you could share like @Viicos suggests.

**Comment by sevetseh28** [comment]
Hey! thanks for your replies. I'm using v 2.8.2.

The problem is happening only when I use `model_dump`, when I call `model_dump_json` it works well and it serialises it.

@ecly I tried your example and it works even with model_dump.

I can't share exactly my models but I will try to reproduce it and share it here.

**Comment by sevetseh28** [comment, marked as answer]
Ok I found the issue, 

I was later using`json.dumps` directly with the `model_dump` dict result (containing `PosixPath` objects) which is using the default JSON encoder. I reckon I gotta use `model_dump_json` directly. 

Thanks for the quick replies!
